core/src/main/java/org/apache/calcite/sql/SqlDeclareHandler.java

```diff
@@ -86,6 +90,21 @@ public SqlDeclareHandler(SqlParserPos pos, HandlerType handlerType,
     }
   }
 
+  @Override public void validate(SqlValidator validator,
+      SqlValidatorScope scope) {
+    BlockScope bs = (BlockScope) scope;
```

Comment:
Do you need to check that this is a valid cast?

Reply:
Yeah I think that would be good to check, I'll update this